akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -389,6 +389,14 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
    */
   def recoveryFinished: Boolean = !recoveryRunning
 
+  override def stash(): Unit = {
+    context.asInstanceOf[ActorCell].currentMessage match {
+      case Envelope(_: JournalProtocol.Response, _) ⇒
+        throw new IllegalStateException("Do not call stash inside of persist callback.")
```

Comment:
will this catch the case when stash is called during recovery also? then the err msg should "... or during recovery"